ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,376 @@
+        public PsychSignalDataConverter(string sourceDirectory, string destinationDirectory, string knownTickerFolder)
+        {
+            _rawSourceDirectory = new DirectoryInfo(sourceDirectory);
+            _destinationDirectory = new DirectoryInfo(destinationDirectory);
+
+            _fileHandles = new Dictionary<string, TickerData>();
+            _knownTickers = Directory.GetDirectories(knownTickerFolder)
+                .Select(Path.GetFileName)
+                .ToHashSet();
+
+            _destinationDirectory.Create();
+        }
+
+        /// <summary>
+        /// Converts a specific file to Lean alternative data format. Note that you must flush
+        /// after you're done converting a file to ensure that all data gets written to disk.
+        /// You can do that by calling <see cref="Dispose"/> once you've finished processing
+        ///
+        /// Note: Assumes that it will be given files in ascending order by date
+        /// </summary>
+        /// <param name="sourceFilePath">File to process and convert</param>
+        public void Convert(FileInfo sourceFilePath)
+        {
+            if (_disposedValue)
+            {
+                throw new ObjectDisposedException("PsychSignalDataConverter has already been disposed");
+            }
+
+            Log.Trace($"PsychSignalDataConverter.Convert(): Begin converting {sourceFilePath.FullName}");
+
+            foreach (var line in File.ReadLines(sourceFilePath.FullName))
+            {
+                var csv = line.Split(',');
+                var ticker = csv[1].ToLower();
+                DateTime timestamp;
+
+                if (csv[0] == "SOURCE" ||
+                    !_knownTickers.Contains(ticker) ||
+                    !DateTime.TryParseExact(csv[2], @"yyyy-MM-dd\THH:mm:ss\Z", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out timestamp))
+                {
+                    continue;
+                }
+
+                TickerData handle;
+                if (!_fileHandles.TryGetValue(ticker, out handle))
+                {
+                    handle = new TickerData(ticker, timestamp.Date, _destinationDirectory);
+                    _fileHandles[ticker] = handle;
+                }
+
+                handle.Append(timestamp, csv);
+            }
+
+            Log.Trace($"PsychSignalDataConverter.Convert(): Finished converting {sourceFilePath.FullName}");
+        }
+
+        /// <summary>
+        /// Filter raw files to be inclusive within the date range specified and converts those files.
+        /// </summary>
+        /// <param name="startDateUtc">Date to start parsing files from</param>
+        /// <param name="endDateUtc">Date to stop parsing files</param>
+        /// <remarks>Inclusive on the lower bound, and exclusive on the upper bound</remarks>
+        public void ConvertFrom(DateTime startDateUtc, DateTime endDateUtc)
```

Comment:
In general the data processing unit is a single date. Therefore, implemet a convert method for a single date, maybe an overload?